src/classes/ADDR_Validator_TDTM.cls

```diff
@@ -37,59 +37,72 @@
+    * @description To support DML operations with more than 200 records, use a static Set of Id's that
+    * have been passed to the verifyAddressRecords() method. This allows for a recursion check that goes
+    * by the Id's of the records as opposed to a simple boolean.
     */
-	public class ADDR_Validator_Exception extends Exception { }
-	
-	/*******************************************************************************************************
-    * @description Indicates whether the class has been run in the current context.
+    public static Set<Id> processedIdsForValidation = new Set<Id>();
+
+    /*******************************************************************************************************
+    * @description Exception raised during the address validation process.
     */
-	public static Boolean alreadyRun = false;
-	
-	/*******************************************************************************************************
+    public class ADDR_Validator_Exception extends Exception { }
+
+    /*******************************************************************************************************
     * @description Verifies new or modified address records, if verification enabled in the settings.
     * @param listNew The list of Addresses from trigger new. 
     * @param listOld The list of Addresses from trigger old. 
     * @param triggerAction The trigger event (BeforeInsert, AfterInsert, etc.) that caused the class to run. 
     * @param objResult The describe for Addresses.
     * @return dmlWrapper The wrapper with the records to perform DML on. 
     */
-	public override DmlWrapper run(List<SObject> newList, List<SObject> oldList, 
-    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) { 
-        
-        if(!alreadyRun) {
-	        Addr_Verification_Settings__c settings = UTIL_CustomSettingsFacade.getAddressVerificationSettings();
-	        if ((triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) &&
-	           settings.enable_automatic_verification__c) {        	
-	        	for(SObject so : newList) {
-	        		Address__c address = (Address__c)so;
-	        		if(!system.isFuture() && !system.isBatch()) {
-	        		    address.Verification_Status__c = Label.Addr_Pending_Verification;
-	        		} else {
-	        			address.Verification_Status__c = Label.Addr_Verification_Required;
-	        		}	
-        	    }
-	        }
+    public override DmlWrapper run(List<SObject> newList, List<SObject> oldList,
+    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        // Look in the static list of previoysly processed Id's to determine if the record has already
+        // gone through this trigger. If so, then don't process it a second time.
+        Boolean recordsPreviouslyProcessed = false;
+        if (newList[0].Id != null && processedIdsForValidation.contains(newList[0].Id)) {
```

Comment:
That is true for delete triggers, though this particular trigger isn't called on a delete. Still, there might be some point in the future where it is, so it's a good idea to add the additional check.